src/test/java/jUnit/JsonParserTest.java

```diff
@@ -1,4 +1,83 @@
 package jUnit;
 
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import shop.Cart;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class JsonParserTest {
+
+    Cart initialCart;
+    JsonParser parser;
+
+    List<String> fileNameList = Stream.of(
+            "vadim-cart",
+            "vadim1-cart",
+            "vadim2-cart")
+            .collect(Collectors.toList());
+
+    @BeforeAll
+    void setUp() {
+        parser = new JsonParser();
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {0, 1, 2})
+    void writeToFileTest(int i) {
+        initialCart = new Cart(fileNameList.get(i));
+        parser.writeToFile(initialCart);
+        Cart actualCart = parser.readFromFile(new File("src/main/resources/" + fileNameList.get(i) + ".json"));
+        assertAll(() -> assertEquals(initialCart.getCartName(), actualCart.getCartName()),
```

Comment:
good practice would be to 
a) write data using a parser but then read data using custom Java code 
b) write data to file using custom Java code and then read using a parser

in that case you will be 100% sure all operations work correctly, because we might face a situation when there is a bug in both write and read operations